src/basic/glob-util.c

```diff
@@ -58,6 +58,14 @@ int glob_exists(const char *path) {
                 return false;
```

Comment:
should do `if (ret_first) *ret_first = NULL;` here...

after all we have this rule that on success we should initialize all return vlaues, and on error leave them unset. But failure in this sense is only stuff < 0, success is everything >= 0.